index.js

```diff
@@ -27,21 +27,37 @@ module.exports = {
     return this.addonConfig.namespacing !== false;
   },
 
+  _isAddon: function() {
+    return Boolean(this.parent.parent);
+  },
+
+  _allPodStyles: [],
+
   included: function(app) {
     if (app.app) { app = app.app; }
 
     this._super.included.apply(this, arguments);
 
     this.projectRoot = app.trees.app;
+
+    if (this._isAddon()) {
+      this.parent.treeForMethods['addon-styles'] = 'treeForParentAddonStyles';
+      this.parent.treeForParentAddonStyles = this.treeForParentAddonStyles.bind(this);
+      this.projectRoot = this.parent.root + '/addon';
+    }
+
     this.appConfig = app.project.config(app.env);
     this.addonConfig = this.appConfig['ember-component-css'] || {};
     this.allowedStyleExtensions = app.registry.extensionsForType('css').filter(Boolean);
   },
 
   treeForAddon: function(tree) {
     if (this._namespacingIsEnabled()) {
-      var podStyles = this._getPodStyleFunnel();
-      var podNames = new ExtractNames(podStyles, {
+      var allPodStyles = new Merge(this._allPodStyles, {
+        annotation: 'Merge (ember-component-css merge all process styles for a complete list of styles)'
```

Comment:
After some digging (using this in an in-repo addon), I found I needed to add `overwrite: true` to this, because it was picking up my main project twice. But this feels a little wrong, and maybe it should be checking to make sure that `processComponentStyles` doesn't add a tree twice?